Seasons Defense/Assets/Scripts/EnemySpawner.cs

```diff
@@ -26,19 +31,32 @@ private IEnumerator SpawnEnemies(float interval, GameObject[] enemy)
         LevelManager.Instance.finishedSpawningEnemies = false;
         LevelManager.Instance.enemiesCount = 0;
         
-        while (enemyCount > 0)
+        while(waveCount > 0)
         {
-            LevelManager.Instance.enemiesCount++;
-
-            //Chooses a random Spawn Point location for the enemy
-            int randomSpawnPoint = Random.Range(0, spawnLocations.Length);
-            int i = Random.Range(0, enemy.Length);
-            Instantiate(enemy[i], spawnLocations[randomSpawnPoint].position, Quaternion.identity);
-            yield return new WaitForSeconds(interval);
-            enemyCount--;
+            waveCount--;
+            while (enemyCount > 0)
+            {
+                LevelManager.Instance.enemiesCount++;
+
+                //Chooses a random Spawn Point location for the enemy
+                int randomSpawnPoint = Random.Range(0, spawnLocations.Length);
+                int i = Random.Range(0, enemy.Length);
+                Instantiate(enemy[i], spawnLocations[randomSpawnPoint].position, Quaternion.identity);
+                yield return new WaitForSeconds(interval);
+                enemyCount--;
+            }
+
+            enemyCount = enemyAtStart + 1;
+            yield return new WaitForSeconds(9);
+            LevelManager.Instance.finishedSpawningEnemies = true;
```

Comment:
Should this line go outside of the outer while loop? Since the spawning of enemies could go on for another wave after this assignment